### DISCUSSION TITLE
How get the list of all the models an all method

### DISCUSSION THREAD

**Discussion by fgvicente** [loopbackio/loopback-next#7180] [Q&A]
Ok. 
in lb3...I get al model with this.... 

` let models = app.models();`

and I get all method for one model this...

` let model = app.models[modelName];
    let methods = model.sharedClass.methods()`

my question...
How would it be in lb4?

**Comment by jannyHou** [comment]
@fgvicente Models in LoopBack 4 describe the shape of data. Behavior like CRUD operations is added by repositories. This is different from LoopBack 3.x where models implement behavior too. You can check [model](https://loopback.io/doc/en/lb4/Model.html) and [migrating models from LB3 to LB4](https://loopback.io/doc/en/lb4/migration-models-overview.html) to understand more about the differences.

> in lb3...I get all model with this....
let models = app.models();

In LB4 your models are usually defined in folder `src/models`, e.g. in application.ts file, you can import all model classes as 

```ts
import * as models from './models';
```

The APIs are defined in repositories, see https://loopback.io/doc/en/lb4/Repositories.html

**Comment by deepakrkris** [comment]
@fgvicente all models are binded in the application context with prefix `model.`, you can get a model with ctx.get('models.' + modelName). Please take a look at https://github.com/strongloop/loopback-next/pull/5378

**Comment by raymondfeng** [comment]
A better way is to use `ctx.findByTag('model')`

**Comment by Jackclarify** [comment]
Try this please:
const lb3App = require('../lb3app/server/server');
const models = lb3App.models; let methods = model.sharedClass.methods();
[Reference](https://github.com/loopbackio/loopback-next/blob/master/examples/lb3-application/lb3app/test/authentication.js#L41)
